File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/PublicConstantField.vb
Imports System

Namespace Tests.Diagnostics
    Public Class A
        Public Const A As Integer = 5 ' Noncompliant {{Change this constant to a 'Shared Read-Only' property.}}
'                    ^
        Private Const B As Integer = 5
        Public C As Integer = 5
    End Class

    Friend Class B
        Public Const A As Integer = 5
        Private Const B As Integer = 5
        Public C As Integer = 5
    End Class
End Namespace

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/PublicConstantField.cs
namespace Tests.Diagnostics
{
    public class A
    {
        public const int A1 = 5; // Noncompliant {{Change this constant to a 'static' read-only property.}}
//                       ^^
        private const int B = 5;
        public int C = 5;
    }

    internal class b
    {
        public const int A = 5; // Compliant
        private const int B = 5;
    }
}
